sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts

```diff
@@ -146,7 +150,11 @@ export class SchemaRegistryClient implements SchemaRegistry {
         (paramOptions) => this.client.schema.getById(id, paramOptions),
         options || {}
       );
-      return convertSchemaResponse(flatResponse, rawResponse);
+      const schema = convertSchemaResponse(flatResponse, rawResponse);
+      // we can not cache the other way because we do not have enough information
+      // about the schema description such as the name and the group
```

Comment:
I think we should raise this with the service team. I believe this was discussed at one point. IIRC, there's a location header that comes back that includes the group and name, though we wouldn't want to be parsing that. I think it would make sense for the service to separate these out in their own headers. cc @hmlam

Reply:
Ok, I am parsing the location header for this release and opened https://github.com/Azure/azure-sdk-for-js/issues/16763 to track the effort of making the service return this info in separate headers.

Reply:
noted. we can add that as additional header(s) on response. cc @nick

Reply:
I'm not sure if it's worth bothering the parsing in the meantime. Seems fragile to me. I would consider leaving it as you had it and adding the other direction when the service gives us the data in an easy to consume way.

Reply:
But I'll leave it to you to make the call. I think either way is fine if we're tracking the follow-up.